scheds/rust/scx_layered/src/main.rs

```diff
@@ -1990,17 +2000,39 @@ impl<'a> Scheduler<'a> {
             debug!("calling NVML to update GPU pids");
             self.gpu_mon_data.last_nvml_poll = std::time::SystemTime::now();
             let device_count = nvml.device_count()?;
-
             let mut pids = HashSet::new();
-
             // Iterate over all devices and gather running processes
             for i in 0..device_count {
                 let device = nvml.device_by_index(i)?;
+                // this feels expensive and static, so do it once.
+                if self.gpu_mon_data.gpu_to_affinity.is_empty() {
+                    let affinity = device.cpu_affinity(CPUMASK_MAX_LONGS)?;
+                    let mut cpu_set = CpuSet::new();
+                    for (x, &sub_cpumask) in affinity.iter().enumerate() {
+                        for cpu_index in 0..64 {
+                            if ((sub_cpumask >> cpu_index) & 1) == 1 {
+                                cpu_set.set(x * 32 + cpu_index)?;
+                            }
+                        }
+                    }
+                    self.gpu_mon_data.gpu_to_affinity.insert(i, cpu_set);
+                }
 
                 let procs = device.running_compute_processes()?;
 
                 for proc_info in procs {
                     pids.insert(proc_info.pid);
+                    if self.opts.affinitize_gpu_tasks
+                        && !self.gpu_mon_data.gpu_to_affinity.is_empty()
+                    {
+                        let cpu_set = self.gpu_mon_data.gpu_to_affinity.get(&i);
+                        if let Some(cpu_set) = cpu_set {
+                            sched_setaffinity(
+                                nix::unistd::Pid::from_raw(proc_info.pid.try_into()?),
+                                cpu_set,
+                            )?;
```

Comment:
Is this a good idea, especially in layered? As I understand it any task with affinity != a NUMA node will go into the lo_fallback_dsq, which should have the effect of deprioritising these tasks. These are likely to be NUMA aligned but I don't see any checks here, or a comment saying they will always be NUMA aligned (I don't think that's possible, especially with AMD's ignore NUMA BIOS feature). If they're NUMA aligned the layer they fall into needs to be Open, have `allow_node_aligned` set, and the machine can't saturate for risk of stalls.

I think some additional checks or warnings here would be a good idea, even if this is behind a flag.